### ISSUE TITLE
Hex Address for audio_parameter_sound_cabrio_processing

### ISSUE THREAD

**Issue by Jay-T88** [NumberOneBot/mqb-mib-sound-datasets#4]
Hi Alex, your work is just awesome and I used that already to linearize MIB2 in my Audi TT Roadster by changing 0x003000. It works just great. To my surprise sound changes completely when I open the roof. So it seems that there are more sound sets within MIB that needs to be tweaked for "roof open". As I've not found any information in the internet yet, I'm happy to do some research with my MIB2 unit to help the community. Just need some guidance from your side: Any ideas for hex-address to search for? I'm happy to share a backup of my unit or datasets from 5F_MIB2...

**Comment by NumberOneBot** [comment]
Hi. Its a pleasure to hear that my work has been helpful to someone. What changes did you make and how does it sound now? 
**audio_parameter_sound_cabrio_processing** has 0x003900 address and I'm looking into it right now. Should be an easy task.

**Comment by NumberOneBot** [comment]
Pushed update with 0x003900 template a moment ago. Took me ~40 minutes :)

**Comment by Jay-T88** [comment]
Wow Alex... that's amazing how fast you've adopted that and created a 010 Editor Hex Editor Template.
To answer you question: As I've installed an external amplifier to the MMI High level outputs, my goal was to minimize any internal sound processing by MMI (see screenshot) I've used Car Scanner App to read 5F 0x003000 dataset and then used your template to edit and put all gain values to zero -> so flat and write back via Car Scanner!
Effect is that music sounds so much better, clearer and more dynamic -> just great thanks to your prework :)

![MIB2 5F 0x003000 Dataset original](https://github.com/NumberOneBot/mqb-mib-sound-datasets/assets/132943993/36827e5f-7b3f-44eb-9312-4c403c17c4ce)

**Comment by Jay-T88** [comment]
oooh I've just realized that Car Scanner Pro is not yet able to read 5F 0x003900 dataset... need to find a workaround with M.I.B or a similar Tool ...

**Comment by Jay-T88** [comment]
Quick feedback: template seems ok but I had to add "<optimize=true>" to 2 lines 196+197 to get it work properly:
CarEQ Profile_1[6] <optimize=true>;
CarEQ Profile_2[6] <optimize=true>;
Will write back modified Dataset and check if this solved my point

**Comment by NumberOneBot** [comment]
Thx for pointing to a bug, It was fast and dirty solution :)

Did you try to null the compression?

**Comment by Jay-T88** [comment]
> Did you try to null the compression? 

No not yet, kept stock values. Does it make a big difference? Have you tried that?

Struggling to write back 0x003900 as I don't have VCP/ODIS... asked Mr-MIBonk if I could write back the MU1427-datasets.csv with a script without doing a full restore of MMI MIB2.

**Comment by NumberOneBot** [comment]
> No not yet, kept stock values. Does it make a big difference? Have you tried that?

Not yet. But it should have a big impact to the sound clarity (in theory).

**Comment by NumberOneBot** [comment]
There is also one more (kind of) dangerous parameter - THDmax. Potentially could burn the amp on higher volumes, but will make it not going to protection, but doing everything it can.

**Comment by NumberOneBot** [comment]
We found it! 0x003900 dataset for cabrio / roadster cars located at 3221422104, 0xC0030018 address in M.I.B. dumps

**Comment by Jay-T88** [comment]
> We found it! 0x003900 dataset for cabrio / roadster cars located at 3221422104, 0xC0030018 address in M.I.B. dumps

Thanks Alex... I can confirm that... I've found the others as well in MUxxxx-datasets.csv M.I.B-Backup:
0xC0030014 = 0x003000 -> Audio Sound Param. EQ Details closed roof
0xC0030018 = 0x003900 -> Audio Sound Param. EQ Details open roof (Cabrio)
0xC0030020 = 0x003B00 -> Audio Sound Param. Config. for Knobs/Vol/Balance

I've modified the areas already and will do a complete restore the weekend as I don't have ODIS/VCP to just write 0x003900 only

**Comment by NumberOneBot** [comment]
M.I.B. guys are already working on update which will allow single dataset uploads / backups :)

**Comment by Jay-T88** [comment]
That's great news... This guy "ivan13" has found some more addresses and put a list together "datasets.pdf" - may be useful for the community: https://www.digital-eliteboard.com/threads/aktivierung-des-subwoofer-menues-in-on-mhi2q-nicht-fuer-b-o.507897/post-4084447

**Comment by NumberOneBot** [comment]
https://github.com/NumberOneBot/mqb-mib-sound-datasets/blob/main/5F_MIB2-datasets.txt

this list was there all the time )

**Comment by Jay-T88** [comment]
ah great haven't seen that before - so there is the script to write dataset without ODIS/VCP - thank you very much.. that was what I was looking for... !!! THANK YOU!!!

**Comment by Jay-T88** [comment]
Alex I just tried your way with individual command script + M.I.B to write modified dataset to 0xC0030018
It seemed to work, but need to check during the day with loud music. I've also nulled DRC, curios if I can hear impact.
Meanwhile I had DTC Errors after running the script - tried to fix SVM but seems like my FW is not supported yet.
I have asked M.I.B. guys for a patch or FW update to supported version.

5F Electronic Information System 15F. Electronic Information System 1
----------------------------
01600 [0x000640] - B201200: Data set implausible
01555 [0x000613] - B201A00: Checking software version management
